src/test/java/picard/sam/markduplicates/UmiAwareMarkDuplicatesWithMateCigarTest.java

```diff
@@ -322,14 +331,84 @@ public void testUmiMetrics(List<String> umis, List<String> assignedUmi, final Li
         UmiAwareMarkDuplicatesWithMateCigarTester tester = getTester(false);
         tester.addArg("MAX_EDIT_DISTANCE_TO_JOIN=" + editDistanceToJoin);
 
-        for( int i = 0;i < umis.size();i++ ) {
-            tester.addMatePairWithUmi(umis.get(i), assignedUmi.get(i), isDuplicate.get(i), isDuplicate.get(i));
+        for (int i = 0;i < umis.size();i++) {
+            tester.addMatePairWithUmi("A", umis.get(i), assignedUmi.get(i), isDuplicate.get(i), isDuplicate.get(i));
         }
         tester.setExpectedAssignedUmis(assignedUmi);
         tester.setExpectedMetrics(expectedMetrics);
         tester.runTest();
     }
 
+    @DataProvider(name = "testMultipleLibraryUmiMetricsDataProvider")
+    private Object[][] testMultipleLibraryUmiMetricsDataProvider() {
+
+        final double effectiveLength2_1 = -(2./3.)*Math.log(2./3.)/Math.log(4.) -(1./3.)*Math.log(1./3.)/Math.log(4.);
+        final double effectiveLength1_1_1 = -3*(1./3.)*Math.log(1./3.)/Math.log(4.);
+        final double estimatedBaseQuality1_12 = QualityUtil.getPhredScoreFromErrorProbability(1./12.);
+
+        return new Object[][] {{
+                // Test basic error correction using edit distance of 1
+                Arrays.asList("A", "B", "A", "B", "A"),
```

Comment:
This could pass array types directly to the test and avoid the `asList()` here.

Although the way this is used, you don't even need to use a testng dataprovider as you only call the test method once with a single set of parameters.